DuckDuckGo/MacBrowserWaitlistView.swift

```diff
@@ -0,0 +1,465 @@
+//
+//  MacBrowserWaitlistView.swift
+//  DuckDuckGo
+//
+//  Copyright © 2022 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+
+typealias ViewActionHandler = (MacWaitlistViewModel.ViewAction) -> Void
+
+// swiftlint:disable file_length
+struct MacBrowserWaitlistView: View {
+
+    @EnvironmentObject var viewModel: MacWaitlistViewModel
+    
+    var body: some View {
+        switch viewModel.viewState {
+        case .notJoinedQueue:
+            MacBrowserWaitlistSignUpView(requestInFlight: false,
+                                         showNotificationAlert: $viewModel.showNotificationPrompt) { action in
+                Task { await viewModel.perform(action: action) }
+            }
+        case .joiningQueue:
+            MacBrowserWaitlistSignUpView(requestInFlight: true,
+                                         showNotificationAlert: $viewModel.showNotificationPrompt) { action in
+                Task { await viewModel.perform(action: action) }
+            }
+        case .joinedQueue(let state):
+            MacBrowserWaitlistJoinedWaitlistView(notificationState: state,
+                                                 showNotificationAlert: $viewModel.showNotificationPrompt) { action in
+                Task { await viewModel.perform(action: action) }
+            }
+        case .invited(let inviteCode):
+            MacBrowserWaitlistInvitedView(inviteCode: inviteCode,
+                                          activityItems: viewModel.createShareSheetActivityItems(),
+                                          showShareSheet: $viewModel.showShareSheet) { action in
+                Task { await viewModel.perform(action: action) }
+            }
+        }
+    }
+
+}
+
+struct MacBrowserWaitlistSignUpView: View {
+
+    let requestInFlight: Bool
+    @Binding var showNotificationAlert: Bool
+    
+    let action: ViewActionHandler
+
+    var body: some View {
+        GeometryReader { proxy in
+            ScrollView {
+                VStack(alignment: .center, spacing: 8) {
+                    HeaderView(imageName: "MacWaitlistJoinWaitlist", title: UserText.macWaitlistTryDuckDuckGoForMac)
+                    
+                    Text(UserText.macWaitlistSummary)
+                        .font(.custom("proximanova-regular", size: 16))
```

Comment:
Lotta custom font usage in here that needs to be cleaned up.

Reply:
could add a Font extension:

    extension Font {
        static func proximaNova(size, weight, etc) -> Self {
        }
    }

Note a blocker for me though.

What happens if Proxima nova isn't available (ie for other developers)?

Reply:
Done ✅ I also checked what happens when the font can't be found, it falls back to the system font and looks pretty good.

Reply:
Example of when the font is missing: ![Proxima Nova Missing](https://user-images.githubusercontent.com/183774/158927428-df909948-1e68-45bc-b898-51cc0e373663.png)